clippy_lints/src/string_as_str.rs

```diff
@@ -0,0 +1,98 @@
+use clippy_utils::{diagnostics::span_lint_and_sugg, ty::is_type_diagnostic_item};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, QPath};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
```

Comment:
Can this be moved to `Method` lint pass? This commit is helpful: https://github.com/rust-lang/rust-clippy/commit/8acc4d2f1e255e0e97e48b7a285a7d6db4a03bd6

Reply:
Done :)